source/cellular_pktio.c

```diff
@@ -1046,7 +1064,11 @@ static void _pktioReadThread( void * pUserData )
                 /* Keep Reading until there is no more bytes in comm interface. */
                 do
                 {
+                    /* When pktio thread is parsing the response from modem, pktio
+                     * thread holds the PktRespMutex until no RX data from comm interface. */
+                    PlatformMutex_Lock( &pContext->PktRespMutex );
                     bytesRead = _handleRxDataEvent( pContext, &pAtResp );
```

Comment:
Lock whole rx event here. 
But we need to call callback functions to users and they're not handled by ourselves.
Is it possible to decouple them?